l1-contracts/deploy-scripts/LensScript.s.sol

```diff
@@ -0,0 +1,371 @@
+import {ITransparentUpgradeableProxy} from "@openzeppelin/contracts-v4/proxy/transparent/TransparentUpgradeableProxy.sol";
+import {Ownable2Step} from "@openzeppelin/contracts-v4/access/Ownable2Step.sol";
+import {TransitionaryOwner} from "../contracts/governance/TransitionaryOwner.sol";
+import {IAdmin} from "../contracts/state-transition/chain-interfaces/IAdmin.sol";
+import {Diamond} from "../contracts/state-transition/libraries/Diamond.sol";
+import {Migrator, MigrationParams} from "../contracts/upgrades/Migrator.sol";
+import {IGetters} from "../contracts/state-transition/chain-interfaces/IGetters.sol";
+
+struct Config {
+    uint256 lensChainId;
+    address lensDiamondProxy;
+    address baseToken;
+    address tempProxyAdmin;
+    address tempBridgehub;
+    address tempStateTransitionManager;
+    address tempValidatorTimelock;
+    address tempL1SharedBridge;
+    address tempErc20Bridge;
+    address newStateTransitionManager;
+    address newBridgehub;
+    address newProxyAdmin;
+    address newValidatorTimelock;
+    address newVerifier;
+}
+
+contract LensScript is Script {
+    using stdToml for string;
+
+    bytes32 public constant PROXY_ADMIN_SLOT = 0xb53127684a568b3173ae13b9f8a6016e243e63b6e8ee1178d6a717850b5d6103;
+
+    Config public config;
+
+    IGovernance.Call[] internal calls;
+    bytes public tempEcoScheduleOperation;
+    bytes public tempEcoExecuteOperation;
+    bytes public vtOwnershipCall;
+
+    address public migrationUpgradeAddress;
+
+    address public transitionaryOwner;
+
+    function setMigrationUpgradeAddress(address _migrationUpgradeAddress) external {
+        migrationUpgradeAddress = _migrationUpgradeAddress;
+    }
+
+    function setTransitionaryOwner(address _transitionaryOwner) external {
+        transitionaryOwner = _transitionaryOwner;
+    }
+
+    function run() external {
+        _initializeConfig();
+        // generate call for transitionary owner to transfer ownership
+        // create gov ops to transfer ownership of temp contracts to transitionary owner
+        //  - schedule
+        //  - execute
+        _generateTempEcoOwnershipCalls();
+        // generate call to accept ownership
+        _generateAcceptOwnershipCalls();
+        // register chain with stm
+        _generateRegisterChainCall();
+        // create chain with bh
+        _generateCreateChainCall();
+        // execute upgrade on lens diamond
+        _generateExecuteUpgradeCall();
+
+        // print generated calldata
+        // _printTempGovernanceScheduleOperation();
+        // _printTempGovernanceExecuteOperation();
+        // _printMigrationExecuteUpgradeCall();
+    }
+
+    function _initializeConfig() internal {
+        string memory root = vm.projectRoot();
+        string memory path = string.concat(root, "/script-config/lens-migration.toml");
+        string memory toml = vm.readFile(path);
+
+        // migrationUpgradeAddress = toml.readAddress("$.migration_upgrade_address");
```

Comment:
I understand that it is easier to test this way, but it is better to have a script that can be reviewed (i.e. tested) and run by anyone, I think it is not too hard in this case, so I guess these addresses will have to be read from the config (or deployed as a part of the script)